sklearn/kernel_approximation.py

```diff
@@ -32,6 +36,37 @@
 )
 
 
+# Utility Functions
+def _return_float_dtype(X, Y):
+    """
+    1. If dtype of X and Y is float32, then dtype float32 is returned.
+    2. Else dtype float is returned.
+    """
+    if not sp.issparse(X) and not isinstance(X, np.ndarray):
+        X = np.asarray(X)
+    if Y is None:
+        Y_dtype = X.dtype
+    elif not sp.issparse(Y) and not isinstance(Y, np.ndarray):
```

Comment:
codecov is complaining about this case not being covered, and I pulled this straight out of [pairwise.py](https://github.com/scikit-learn/scikit-learn/blob/e3b383a2d1d2703560a63d0eb775130fc3584e4e/sklearn/metrics/pairwise.py#L50).... so I wonder if we should add this in `sklearn/utils/_array_api.py`?